sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BigDecimalCoder.java

```diff
@@ -55,7 +55,8 @@ public void encode(BigDecimal value, OutputStream outStream) throws IOException,
   @Override
   public void encode(BigDecimal value, OutputStream outStream, Context context)
       throws IOException, CoderException {
-    checkNotNull(value, String.format("cannot encode a null %s", BigDecimal.class.getSimpleName()));
+    checkArgumentNotNull(
```

Comment:
Since this is actually a hot path, it would be good to avoid the string formatting like this:

`checkArgumentNotNull(value, "cannot encode a null %s", BigDecimal.class.getSimpleName())`

It seems that there are many opportunities for this.

I know this is already in the code. You could make them separate commits since they are independent changes. Or you can skip that and someone else can do it later.